src/com/koxudaxi/poetry/PyAddExistingPoetryEnvPanel.kt

```diff
@@ -55,20 +58,32 @@ class PyAddExistingPoetryEnvPanel(private val project: Project?,
                 .panel
         add(formPanel, BorderLayout.NORTH)
         addInterpretersAsync(sdkComboBox) {
-            detectPoetryEnvs(module, existingSdks, context, project?.basePath ?: newProjectPath)
+            val existingSdkPaths = sdkHomes(existingSdks)
+            val moduleSdks = allModules(project).parallelStream().flatMap { module ->
+                val sdks = detectPoetryEnvs(module, existingSdkPaths, module.basePath)
+                        .filterNot { it.isAssociatedWithAnotherModule(module) }
+                sdks.forEach { sdkToModule.putIfAbsent(it.name, module) }
+                sdks.stream()
+            }.toList()
+            val rootSdks = detectPoetryEnvs(module, existingSdkPaths, project?.basePath ?: newProjectPath)
                     .filterNot { it.isAssociatedWithAnotherModule(module) }
+            val moduleSdkPaths = moduleSdks.map { it.name }.toSet()
+            val sdks = rootSdks.filterNot { moduleSdkPaths.contains(it.name) } + moduleSdks
+            sdks.sortedBy { it.name }
         }
     }
 
     override fun validateAll(): List<ValidationInfo> = listOfNotNull(validateSdkComboBox(sdkComboBox, this))
 
     override fun getOrCreateSdk(): Sdk? {
         return when (val sdk = sdkComboBox.selectedSdk) {
-            is PyDetectedSdk ->
-                setupPoetrySdkUnderProgress(project, module, existingSdks, newProjectPath,
+            is PyDetectedSdk -> {
+                val mappedModule = sdkToModule[sdk.name] ?: module
```

Comment:
Created sdk matches the corresponding module's name:
<img width="953" alt="Project_Structure_and_data-mesh" src="https://user-images.githubusercontent.com/527952/116505251-69dc5c80-a888-11eb-95d1-b33b4e5fbb32.png">